signing-and-verification-workflow.md

```diff
@@ -0,0 +1,53 @@
+
+## Verification workflow
+
+The user wants to pull an OCI artifact only if they are signed by a trusted publisher and the signature is valid.
+
+### Prerequisites
+
+- User has a fully qualified reference to an OCI artifact they want to pull. If a tag is provided, the tag is looked up and converted to a digest.
+- User has configured [trust store and trust policy](./trust-store-trust-policy-specification.md) required for signature verification.
+
+### Steps
+
+1. **Get signature artifact descriptors:** Using the ORAS [Manifest Referrers API](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md) download the Notary v2 signature artifact descriptors. The ORAS referrers API only needs the artifact digest and `artifactType`(optional) as input. The Notary v2 signature's artifact type is `application/vnd.cncf.notary.v2.signature`.
+1. **Get signature artifact manifests:** For each signature artifact descriptor, perform the following steps:
+    1. **Get signature artifact manifests:** Download the Notary v2 signature's artifact manifests for the given artifact descriptor.
+    1. **Filter signature artifact manifests:**
+        1. Based on the value of the `blobs->mediaType` attribute and supported signature envelope format, filter out signature manifests.
+        1. Depending upon the trust-store configuration, further filter out signature manifests. why? How? **TBD**(See [signature filtering](https://hackmd.io/@pritesh/signature_filtering)).
+        1. If the artifact manifest is filtered out, skip the below steps and move to the next signature artifact descriptor(step 2.1). If all signature artifact descriptors have already been processed, fail the signature verification and exit.
+    1. **Get and validate signatures:** On filtered Notary v2 signature artifact manifest, perform the following steps:
+        1. Download the signature envelope.
+        1. Validate the signature envelope using trust-store and trust-policy as mentioned in [signature evaluation](./trust-store-trust-policy-specification.md#signature-evaluation) section.
+        1. If signature validation fails, skip the below steps and move to the next signature artifact descriptor(step 2.1). If all signature artifact manifests have already been processed, fail the signature verification and exit.
+        1. If signature validation succeeds, compare the given digest with the signed digest present in the signature envelope's payload. If digests are equal, signature verification is considered successful. Otherwise, move to the next signature artifact descriptor(step 2.1). If all signature artifact descriptors have already been processed, fail the signature verification and exit.
```

Comment:
It's unclear what "given digest" mean, can be more descriptive here.

Reply:
Updated.